andela/travel_tool_front
# #160374729 Fix Left Sidebar Bug; Create an Application Shell

#### What does this PR do?
-  Creates an application shell 
  - Avoid re-instantiation of the header bar, left-side navigation bar, notifications pane and the side drawer when navigating the application
  - Fixes unexpected behavior when switching the left sidebar links

#### Description of Task to be completed?
- Create an application shell for constant components
- Fix active link transition bug in the left side navigation bar

#### How should this be manually tested?
- Clone the repo - `https://github.com/andela/travel_tool_front.git`
- Switch to the repo directory - `cd travel_tool_front`
- Checkout into this branch - `git checkout ch-app-shell-`
- Install dependencies and start the app - `make start`
- Navigate to` localhost:3000/`
- Login and navigate the application
  - The constant components should be managing their state properly because they don't re-instantiate
  - The links on the left sidebar should be responding appropriately

#### Any background context you want to provide?
- The bug with the left sidebar is a result of constant components re-instantiating when navigating the application, thus unable to maintain and manage their own local state

#### What are the relevant pivotal tracker stories?
#160374729

#### Screenshots (if appropriate)

![sep-10-2018 08-47-04 - sidebar](https://user-images.githubusercontent.com/23254512/45278534-37fb0980-b4d6-11e8-97cd-c0da0ea6c70f.gif)

#### Questions:
